### PR TITLE
Build system, including templates for PC, Pi and Arduino Due

### PR DESCRIPTION
**Pull request by sqrtroot**
Created templates to start developing for the following targets:

- pc
- pi
- arduino due

building for arduino due on windows pc's requires mingw make

### CODE REVIEW

**Review by ManDeJan — DISMISSED** [review]
Approved, all targets where easy to build, and get running. We should add a wiki page for how to use BMPTK++

**Comment by LuukSteeman** [comment]
@cjsmeele als ik een module aanmaak met pc als target krijg ik geen visual studio solution

**Comment by LuukSteeman** [comment]
nope nvm, verkeerd gekeken
zat in src te kijken niet build, dacht dat ik dat wel gecheckt had tho

**Review by LuukSteeman — CHANGES_REQUESTED** [review]
Add documentation, everything else looks fine

**Comment by cjsmeele** [comment]
@LuukSteeman Thanks for the feedback!

@sqrtroot @ManDeJan, Feature idea: We should probably add a parameter / prompt to allow people to override the cmake generator, so that people can use bmptk++ to generate files for IDEs other than the default for their host platform.

**Comment by LuukSteeman** [comment]
Maybe make it so that double clicking on bmptk++.py works as well. Right now it just opens and closes a cmd window if you do that.
Would be a bit more intuitive than using cmd to run the script.

**Comment by LuukSteeman** [comment]
console works on windows

**Comment by cjsmeele** [comment]
Re Documentation: I'm currently writing [Development Workflow](https://github.com/R2D2-2017/R2D2-2017/wiki/Development-Workflow) on the wiki, which includes instructions on running bmptk++.

**Comment by LuukSteeman** [comment]
allright, notify me when it's done so i can take a look. if that's all good this can be merged

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 2.6)
```
Comment (sqrtroot) [inline CMakeLists.txt]: Moet dit bestand nog bestaan?

File: modules/template-arduino/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+include(../../flags.cmake)
+
+set(sources src/main.cc)
+
+include_directories(
+	${hwlib}/library
+)
+
+set(sources
+    #${hwlib}/library/hwlib.cpp
```
Comment (sqrtroot) [inline modules/template-arduino/CMakeLists.txt]: Moet dit niet wel weer gecompiled worden?

File: modules/template-arduino/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#set(flash_program_path ~/bossac) # Your path to the flash tool
+
+# The compiler that is used:
+#include(${toolchain}/compilers/arm-none-eabi.cmake)
+
+# The target chip that is compiled for:
+include(${toolchain}/targets/arm32/cm3/atmel/sam3x/sam3x8e/sam3x8e.cmake)
+
+# The flashscript used to flash the binary (optional):
+include(${toolchain}/targets/arm32/cm3/atmel/sam3x/sam3x8e/flashscripts/arduino_due.cmake)
```
Comment (sqrtroot) [inline modules/template-arduino/CMakeLists.txt:31]: Waar staat de define voor arduino due?

File: modules/template-pc/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+enable_testing()
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+set(sources
+    src/main.cc)
+
+add_executable(${exe_name} ${sources})
```
Comment (sqrtroot) [inline modules/template-pc/CMakeLists.txt:15]: Flags niet included?

File: modules/template-pi/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+enable_testing()
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+set(sources
+    src/main.cc)
+
+add_executable(${exe_name} ${sources})
```
Comment (sqrtroot) [inline modules/template-pi/CMakeLists.txt:19]: Geen flags included?

**Review by sqrtroot — COMMENTED** [review]
Maybe we have to have a look at these points?

File: modules/template-arduino/CMakeLists.txt (hunk 2 above)
Comment (cjsmeele) [reply modules/template-arduino/CMakeLists.txt]: Nee, zie de wrap-hwlib regel eronder. We includen nergens rechtstreeks hwlib.
Deze regel (en de `set(sources main.cpp)` verder naar boven (waar komt die vandaan?)) kan weg.

**Comment by sqrtroot** [comment]
Documentation is done @ https://github.com/R2D2-2017/R2D2-2017/wiki/Development-Workflow
Small litle things are done please review
@LuukSteeman @ManDeJan

**Comment by LuukSteeman** [comment]
Documentation looks good, good work guys.

**Review by LuukSteeman — APPROVED** [review]

**Pull request merged**